packages/components/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import {
+  Root as SwitchRootPrimitive,
+  Thumb as SwitchThumbPrimitive,
+} from "@radix-ui/react-switch";
+import type { SwitchProps as SwitchPrimitiveProps } from "@radix-ui/react-switch";
+import { useUID } from "react-uid";
+import PropTypes from "prop-types";
+import { Label } from "../Label";
+import { Box } from "../../primitives/Box";
+import { HelpText } from "../HelpText";
+import { Icon } from "../Icon";
+
+export interface SwitchProps extends SwitchPrimitiveProps {
+  /** The controlled state of the switch. Must be used in conjunction with onCheckedChange. */
+  checked?: boolean;
+  /** The accessible text label for the switch. */
+  children: string;
+  /** The state of the switch when it is initially rendered. Use when you do not need to control its state. */
+  defaultChecked?: boolean;
+  /** When true, prevents the user from interacting with the switch. */
+  disabled?: boolean;
+  /** Help text for additional guidance. */
+  helpText?: React.ReactNode;
+  /** When true, indicates that the user must check the switch before the owning form can be submitted. */
+  required?: boolean;
+  /** The name of the switch. Submitted with its owning form as part of a name/value pair. */
+  name?: string;
+  /** Event handler called when the state of the switch changes. */
+  onCheckedChange?: (checked: boolean) => void;
+  /** The value given as data when submitted with a name. */
+  value?: string;
```

Comment:
So we need value and checked then? I'm a little confused with the difference and how this is used.

Reply:
@RohitRajendran the value is there only if you want to pass a custom value if this lives in a form. Otherwise it functions like a checkbox. Say if you want to display content based on if a switch is checked or unchecked.